### PR TITLE
obs-studio: 29.1.3 -> 30.0.2

### PR DESCRIPTION
**Pull request by CobaltCause**
## Description of changes

Attempts to continue https://github.com/NixOS/nixpkgs/pull/269008.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fpletz** [comment]
Things done:
 * squashed OBS bump commits into one
 * added meta to qrcodegencpp from qrcodegen
 * don't install libvpl examples and tools
 * rebased onto master

**Comment by fpletz** [comment]
I'm gonna do some testing and check if all plugins build correctly. Then we should be ready to merge. :beers:

**Comment by fpletz** [comment]
Built all plugins. Fixed some builds and tested those plugins. Also tested basic functionality of plugins I can reasonably test myself without any special hard- or software.

**Comment by CobaltCause** [comment]
I wonder how this PR should relate to https://github.com/NixOS/nixpkgs/pull/264621:

1. Wait for that one to be merged first?
2. Package `libvpl` in this PR and let the other changes in that PR happen later?

I also note that the Nix expression to build `libvpl` is different between these PRs, not sure which one is better just by looking at them...

**Comment by CobaltCause** [comment]
Maybe I'm missing something but it (libvpl and obs-studio) still builds when I apply this patch:

```diff
diff --git a/pkgs/by-name/li/libvpl/package.nix b/pkgs/by-name/li/libvpl/package.nix
index 32ec7bf4fcc9..ef84a9193ce7 100644
--- a/pkgs/by-name/li/libvpl/package.nix
+++ b/pkgs/by-name/li/libvpl/package.nix
@@ -1,11 +1,6 @@
 { stdenv
 , lib
 , fetchFromGitHub
-, wayland
-, wayland-protocols
-, xorg
-, libva
-, libdrm
 , cmake
 , pkg-config
 }:
@@ -26,16 +21,6 @@ stdenv.mkDerivation (finalAttrs: {
     pkg-config
   ];
 
-  buildInputs = [
-    wayland
-    wayland-protocols
-    xorg.libxcb
-    xorg.libX11
-    xorg.libpciaccess
-    libva
-    libdrm
-  ];
-
   cmakeFlags = [
       "-DCMAKE_BUILD_TYPE=Release"
       "-DCMAKE_INSTALL_LIBDIR=lib"
```

This script from upstream seems to imply these dependencies are only needed when building the tools, which we disabled: https://github.com/intel/libvpl/blob/v2.10.1/script/bootstrap#L25-L34

**Comment by fpletz** [comment]
Thanks for the heads up. I wasn't aware of that PR. Yeah, we should get that merged first. It's a shame though that the package init also targets staging since strictly speaking only the ffmpeg change would require that.

I haven't had Intel hardware for a while so I can't really say if temporarily disabling the QuickSync OBS plugin (obs-qsv11) via `-DENABLE_QSV11=OFF` would work for most users. Then we could drop the dependency on `libvpl` for now. My guess would be that encoding on Intel hardware via VAAPI in that case still be available in OBS but I can't test.

**Comment by CobaltCause** [comment]
> It's a shame though that the package init also targets staging since strictly speaking only the ffmpeg change would require that.

Yeah. I wonder if it would be worth making a third PR just to add `libva` that'd get merged before both of the current ones? Could ask how the people active on the other PR feel about that.

> I haven't had Intel hardware for a while so I can't really say if temporarily disabling the QuickSync OBS plugin (obs-qsv11) via `-DENABLE_QSV11=OFF` would work for most users. Then we could drop the dependency on `libvpl` for now. My guess would be that encoding on Intel hardware via VAAPI would still be available in OBS in that case but I can't test.

I have access to an i7-1185G7 and an i7-6700K, so I could help with testing if we wanted to go that route.

File: maintainers/maintainer-list.nix
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:15971]: this should be it's own commit called `maintainers: add running-grass`.

Comment (CobaltCause) [reply maintainers/maintainer-list.nix:15971]: How'd that even get here? Looks like this is from a commit from https://github.com/NixOS/nixpkgs/pull/269098. I'll just drop it from this PR.

Comment (Janik-Haag) [reply maintainers/maintainer-list.nix:15971]: Just do a rebase on master then it should automatically be dropped

Comment (CobaltCause) [reply maintainers/maintainer-list.nix]: Done; thanks for catching this!

File: pkgs/applications/video/obs-studio/plugins/obs-replay-source.nix
Comment (pschmitt) [inline pkgs/applications/video/obs-studio/plugins/obs-replay-source.nix:22]: This won't be necessary anymore with obs-replay-source 1.6.13:

https://github.com/exeldro/obs-replay-source/commit/584c028bca8161efb32d72a0d31ba3071cfad706

Comment (pschmitt) [reply pkgs/applications/video/obs-studio/plugins/obs-replay-source.nix:22]: https://github.com/NixOS/nixpkgs/pull/277396

Comment (fabianhjr) [reply pkgs/applications/video/obs-studio/plugins/obs-replay-source.nix:22]: Merged in https://github.com/NixOS/nixpkgs/pull/277670

Comment (CobaltCause) [reply pkgs/applications/video/obs-studio/plugins/obs-replay-source.nix]: Thanks for pointing this out; I just rebased and dropped the commit that made the changes that are no longer necessary and it indeed builds on my machine.

**Comment by fabianhjr** [comment]
Result of `nixpkgs-review pr 275544` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>39 packages built:</summary>
  <ul>
    <li>libcef</li>
    <li>libvpl</li>
    <li>obs-studio</li>
    <li>obs-studio-plugins.advanced-scene-switcher</li>
    <li>obs-studio-plugins.droidcam-obs</li>
    <li>obs-studio-plugins.input-overlay</li>
    <li>obs-studio-plugins.looking-glass-obs</li>
    <li>obs-studio-plugins.obs-3d-effect</li>
    <li>obs-studio-plugins.obs-backgroundremoval</li>
    <li>obs-studio-plugins.obs-command-source</li>
    <li>obs-studio-plugins.obs-freeze-filter</li>
    <li>obs-studio-plugins.obs-gradient-source</li>
    <li>obs-studio-plugins.obs-gstreamer</li>
    <li>obs-studio-plugins.obs-hyperion</li>
    <li>obs-studio-plugins.obs-livesplit-one</li>
    <li>obs-studio-plugins.obs-move-transition</li>
    <li>obs-studio-plugins.obs-multi-rtmp</li>
    <li>obs-studio-plugins.obs-mute-filter</li>
    <li>obs-studio-plugins.obs-nvfbc</li>
    <li>obs-studio-plugins.obs-pipewire-audio-capture</li>
    <li>obs-studio-plugins.obs-replay-source</li>
    <li>obs-studio-plugins.obs-rgb-levels-filter</li>
    <li>obs-studio-plugins.obs-scale-to-sound</li>
    <li>obs-studio-plugins.obs-shaderfilter</li>
    <li>obs-studio-plugins.obs-source-clone</li>
    <li>obs-studio-plugins.obs-source-record</li>
    <li>obs-studio-plugins.obs-source-switcher</li>
    <li>obs-studio-plugins.obs-teleport</li>
    <li>obs-studio-plugins.obs-text-pthread</li>
    <li>obs-studio-plugins.obs-transition-table</li>
    <li>obs-studio-plugins.obs-tuna</li>
    <li>obs-studio-plugins.obs-vaapi</li>
    <li>obs-studio-plugins.obs-vertical-canvas</li>
    <li>obs-studio-plugins.obs-vintage-filter</li>
    <li>obs-studio-plugins.obs-vkcapture</li>
    <li>obs-studio-plugins.obs-websocket</li>
    <li>obs-studio-plugins.waveform</li>
    <li>obs-studio-plugins.wlrobs</li>
    <li>qrcodegencpp</li>
  </ul>
</details>

**Review by fabianhjr — APPROVED** [review]

**Comment by CobaltCause** [comment]
I suppose I should mention that the author of https://github.com/NixOS/nixpkgs/pull/264621 says [they're okay with the package definition of `libvpl` in this PR](https://github.com/NixOS/nixpkgs/pull/264621#issuecomment-1874833622) which resolves the concern brought up in https://github.com/NixOS/nixpkgs/pull/275544#issuecomment-1865244208.

**Comment by fabianhjr** [comment]
Great, thanks for pointing it out. There are no pending comments and it's ok to merge as is?

**Comment by CobaltCause** [comment]
I think so. There was some desire to include the binaries of `libvpl` in a separate output (https://github.com/NixOS/nixpkgs/pull/264621#issuecomment-1874996483), but I couldn't figure out how to make this work locally (kept hitting errors like `error: cycle detected in build of '...' in the references of output 'bin' from output 'out'`). Maybe someone could make that change later on if it turns out to be needed, but my understanding is that `libvpl` is primarily useful for the library component.

**Review by flexiondotorg — APPROVED** [review]
I tested these changes and its working great.

**Comment by vcunat** [comment]
Patches don't apply: https://hydra.nixos.org/build/245942469

**Pull request merged**
